source/common/upstream/original_dst_cluster.cc

```diff
@@ -93,6 +102,24 @@ HostConstSharedPtr OriginalDstCluster::LoadBalancer::chooseHost(LoadBalancerCont
   return nullptr;
 }
 
+Network::Address::InstanceConstSharedPtr
+OriginalDstCluster::LoadBalancer::requestOverrideHost(LoadBalancerContext* context) {
+  Network::Address::InstanceConstSharedPtr request_host;
+  const Http::HeaderMap* downstream_headers = context->downstreamHeaders();
+  if (downstream_headers &&
+      downstream_headers->get(Http::Headers::get().EnvoyOriginalDstHost) != nullptr) {
+    const std::string& request_override_host =
+        downstream_headers->get(Http::Headers::get().EnvoyOriginalDstHost)->value().c_str();
+    try {
+      request_host = Network::Utility::parseInternetAddressAndPort(request_override_host, false);
+      ENVOY_LOG(debug, "Using request override host {}.", request_override_host);
+    } catch (const Envoy::EnvoyException& e) {
+      ENVOY_LOG(debug, "original_dst_load_balancer: invalid override header value. {}", e.what());
```

Comment:
Sorry which stat is this covered by?

Reply:
There is not stat covering this. I am referring to the existing log line here https://github.com/envoyproxy/envoy/blob/master/source/common/upstream/original_dst_cluster.cc#L92. However I do think, stat may be a good option to add. I have added and adjusted the tests accordingly. I also left couple of TODOs where we can convert the above log line and some other debug log lines to stats in future.